tests/api/api/feature_store/test_feature_sets.py

```diff
@@ -459,3 +464,46 @@ def test_features_list(db: Session, client: TestClient) -> None:
     assert (
         features_response["features"][1]["feature_set_digest"]["metadata"]["tag"] == tag
     )
+
+
+def test_no_feature_leftovers_when_storing_feature_sets(
+    db: Session, client: TestClient
+) -> None:
+    project_name = f"prj-{uuid4().hex}"
+    count = 5
+    name = "feature_set"
+    # Make sure no leftover features remain in the DB after doing multi-store on the same object
+    for i in range(count):
+        feature_set = _generate_feature_set(name)
+
+        _assert_store_feature_set(
+            client, project_name, name, "latest", feature_set, versioned=False
+        )
+        # _feature_set_create_and_assert(client, project_name, feature_set)
+        _assert_list_objects(
+            client, "features", project_name, None, len(feature_set["spec"]["features"])
+        )
+
+    # Now create different features each time we store, make sure no leftovers remain
+    for i in range(count):
+        feature_set = _generate_feature_set(name, f"feature_{i}")
+        _assert_store_feature_set(
+            client, project_name, name, "latest", feature_set, versioned=False
+        )
+        # _feature_set_create_and_assert(client, project_name, feature_set)
+        _assert_list_objects(
+            client, "features", project_name, None, len(feature_set["spec"]["features"])
+        )
+
+    response = client.delete(f"/api/projects/{project_name}/feature-sets/{name}")
+    assert response.status_code == HTTPStatus.NO_CONTENT.value
+
+    # When working on a versioned object, features will be multiplied, since they belong to different versions
+    # (different features change the uid)
+    for i in range(count):
+        feature_set = _generate_feature_set(name, f"feature_{i}")
+        _assert_store_feature_set(
+            client, project_name, name, "latest", feature_set, versioned=True
+        )
+        # _feature_set_create_and_assert(client, project_name, feature_set)
+        _assert_list_objects(client, "features", project_name, None, 4 * (i + 1))
```

Comment:
I'm not sure where this 4 comes from, can you explain ?

Reply:
It's the number of features that are in the created `FeatureSet` object, multiplied by the number of feature-sets actually existing. I see it remained there as a fixed number instead of getting it from the actual objects generated - I'll fix.